frontend/src/app/components/common/comp-coordinate-input.tsx

```diff
@@ -0,0 +1,443 @@
+      }
+    } else {
+      setXCoordinate(input);
+      if (yCoordinate) {
+        handleGeoPointChange(yCoordinate, input);
+      }
+    }
+  };
+
+  const handleGeoPointChange = (latitude: string, longitude: string) => {
+    setYCoordinateErrorMsg("");
+    setXCoordinateErrorMsg("");
+    const regex = /^-?\d+\.?\d*$/;
+    let hasErrors = false;
+    if (!regex.exec(latitude)) {
+      setYCoordinateErrorMsg("Latitude value must be a number");
+      hasErrors = true;
+    }
+    if (!regex.exec(longitude)) {
+      setXCoordinateErrorMsg("Longitude value must be a number");
+      hasErrors = true;
+    }
+    if (latitude && !Number.isNaN(latitude)) {
+      const item = parseFloat(latitude);
+      if (item > bcBoundaries.maxLatitude || item < bcBoundaries.minLatitude) {
+        setYCoordinateErrorMsg(
+          `Latitude value must be between ${bcBoundaries.maxLatitude} and ${bcBoundaries.minLatitude} degrees`,
+        );
+        hasErrors = true;
+      }
+    }
+    if (longitude && !Number.isNaN(longitude) && longitude.trim() !== "-") {
+      const item = parseFloat(longitude);
+      if (item > bcBoundaries.maxLongitude || item < bcBoundaries.minLongitude) {
+        setXCoordinateErrorMsg(
+          `Longitude value must be between ${bcBoundaries.minLongitude} and ${bcBoundaries.maxLongitude} degrees`,
+        );
+        hasErrors = true;
+      } else {
+        syncCoordinates(latitude, longitude);
+      }
+    }
+    throwError(hasErrors);
+  };
+
+  const handleUtmCoordinateChange = (input: string | Option, type: UTMCoordinates) => {
+    if (type === UTMCoordinates.Easting) {
+      setEastingCoordinate(input.toString());
+      if (northingCoordinate && zoneCoordinate) {
+        handleUtmGeoPointChange(input.toString(), northingCoordinate, zoneCoordinate.value as string);
+      }
+    } else if (type === UTMCoordinates.Northing) {
+      setNorthingCoordinate(input.toString());
+      if (eastingCoordinate && zoneCoordinate) {
+        handleUtmGeoPointChange(eastingCoordinate, input.toString(), zoneCoordinate.value as string);
+      }
+    } else {
+      setZoneCoordinate(input as Option);
+      if (northingCoordinate && eastingCoordinate) {
+        handleUtmGeoPointChange(eastingCoordinate, northingCoordinate, (input as Option).value as string);
+      }
+    }
+  };
+
+  const handleUtmGeoPointChange = (easting: string, northing: string, zone: string) => {
+    setEastingCoordinateErrorMsg("");
+    setNorthingCoordinateErrorMsg("");
+    setZoneErrorMsg("");
+    setYCoordinateErrorMsg("");
+    setXCoordinateErrorMsg("");
+
+    const regex = /^\d+\.?\d*$/;
+    let hasErrors = false;
+    let lat;
+    let lng;
+
+    const errorTextSuffix = `The corresponding longitude value must be between ${bcBoundaries.minLongitude} and ${bcBoundaries.maxLongitude} degrees`;
+    const eastingErrorText = `Invalid Easting. ` + errorTextSuffix;
+    const northingErrorText = `Invalid Northing. ` + errorTextSuffix;
```

Comment:
Discussed this in standup, would like to match what is done for Lat/Long e.g.

"Easting value must be between 290220.6 and 720184.9 metres."